library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -26,183 +27,299 @@
-   */
-  @Override public int getItemViewType(int position) {
-    int viewType = TYPE_ITEM;
-    if (isHeaderPosition(position)) {
-      viewType = TYPE_HEADER;
-    } else if (isFooterPosition(position)) {
-      viewType = TYPE_FOOTER;
-    }
-    return viewType;
-  }
-
-  /**
-   * Returns the items list size if there is no a header configured or the size taking into account
-   * that if a header or a footer is configured the number of items returned is going to include
-   * this elements.
-   */
-  @Override public int getItemCount() {
-    int size = items.size();
-    if (hasHeader()) {
-      size++;
-    }
-    if (hasFooter()) {
-      size++;
-    }
-    return size;
-  }
-
-  public H getHeader() {
-    return header;
-  }
-
-  public T getItem(int position) {
-    if (hasHeader() && hasItems()) {
-      --position;
-    }
-    return items.get(position);
-  }
-
-  public F getFooter() {
-    return footer;
-  }
-
-  public void setHeader(H header) {
-    this.header = header;
-  }
-
-  public void setItems(List<T> items) {
-    validateItems(items);
-    this.items = items;
-  }
-
-  public void setFooter(F footer) {
-    this.footer = footer;
-  }
-
-  public void showFooter() {
-    this.showFooter = true;
-    notifyDataSetChanged();
-  }
-
-  public void hideFooter() {
-    this.showFooter = false;
-    notifyDataSetChanged();
-  }
-
-  protected abstract VH onCreateHeaderViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract VH onCreateItemViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract VH onCreateFooterViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract void onBindHeaderViewHolder(VH holder, int position);
-
-  protected abstract void onBindItemViewHolder(VH holder, int position);
-
-  protected abstract void onBindFooterViewHolder(VH holder, int position);
-
-  /**
-   * Returns true if the position type parameter passed as argument is equals to 0 and the adapter
-   * has a not null header already configured.
-   */
-  public boolean isHeaderPosition(int position) {
-    return hasHeader() && position == 0;
-  }
-
-  public boolean isFooterPosition(int position) {
-    int lastPosition = getItemCount() - 1;
-    return hasFooter() && position == lastPosition;
-  }
-
-  /**
-   * Returns true if the view type parameter passed as argument is equals to TYPE_HEADER.
-   */
-  protected boolean isHeaderType(int viewType) {
-    return viewType == TYPE_HEADER;
-  }
-
-  protected boolean isFooterType(int viewType) {
-    return viewType == TYPE_FOOTER;
-  }
-
-  /**
-   * Returns true if the header configured is not null.
-   */
-  protected boolean hasHeader() {
-    return getHeader() != null;
-  }
-
-  /**
-   * Returns true if the footer configured is not null.
-   */
-  protected boolean hasFooter() {
-    return getFooter() != null && showFooter;
-  }
-
-  private boolean hasItems() {
-    return items.size() > 0;
-  }
+        extends RecyclerView.Adapter<VH> {
```

Comment:
You've changed the whole file to use 4 spaces instead of 2. Please apply the checkstyle configuration I commented you few comments ago and reformat the code.